source/common/quic/envoy_deterministic_connection_id_generator.cc

```diff
@@ -0,0 +1,31 @@
+#include "source/common/quic/envoy_deterministic_connection_id_generator.h"
+
+#include "source/common/quic/envoy_quic_utils.h"
+
+namespace Envoy {
+namespace Quic {
+
+absl::optional<quic::QuicConnectionId>
+EnvoyDeterministicConnectionIdGenerator::GenerateNextConnectionId(
+    const quic::QuicConnectionId& original) {
+  auto new_cid = DeterministicConnectionIdGenerator::GenerateNextConnectionId(original);
+  if (!new_cid.has_value()) {
+    return new_cid;
+  }
+  adjustNewConnectionIdForRoutine(new_cid.value(), original);
+  return new_cid;
```

Comment:
Totally you call, but we could save a line by doing:

```
  if (new_cid.has_value()) {
    adjustNewConnectionIdForRoutine(new_cid.value(), original);
  }
  return new_cid;
```

(And below)

Reply:
done